src/inventory.cpp

```diff
@@ -394,11 +394,31 @@ static int count_charges_in_list( const itype *type, const map_stack &items )
 void inventory::form_from_map( const tripoint &origin, int range, bool assign_invlet,
                                bool clear_path )
 {
-    form_from_map( g->m, origin, range, assign_invlet, clear_path );
+    form_from_map( g->m, origin, range, nullptr, assign_invlet, clear_path );
+}
+
+void inventory::form_from_map( const tripoint &origin, int range, player *pl, bool assign_invlet,
+                               bool clear_path )
+{
+    form_from_map( g->m, origin, range, pl, assign_invlet, clear_path );
+}
+
+void inventory::form_from_map( const tripoint &origin, int range, const player *pl,
+                               bool assign_invlet,
+                               bool clear_path )
+{
+    form_from_map( g->m, origin, range, pl, assign_invlet, clear_path );
 }
 
 void inventory::form_from_map( map &m, const tripoint &origin, int range, bool assign_invlet,
                                bool clear_path )
+{
+    form_from_map( m, origin, range, nullptr, assign_invlet, clear_path );
```

Comment:
Instead of making a bunch of new overloads, you could add the `player *` parameter to the existing functions and default it to be `nullptr`.